parsl/tests/test_vineex/test_function_context.py

```diff
@@ -0,0 +1,75 @@
+import os
+
+import pytest
+
+import parsl
+from parsl.app.app import python_app
+
+try:
+    import ndcctools.taskvine  # noqa: F401
+except ImportError:
+    TASKVINE_AVAIL = False
+else:
+    TASKVINE_AVAIL = True
+
+require_taskvine = pytest.mark.skipif(
+        not TASKVINE_AVAIL,
+        reason="Tests use features exclusive to TaskVineExecutor.")
+
+
+def f_context(y, input_file_names):
+    bases = []
+    for file_name in input_file_names:
+        with open(file_name, 'r') as f:
+            bases.append(int(f.read().strip()))
+
+    return {'y': sum(bases) + y * 2}
+
+
+@python_app
+def f_compute(x, parsl_resource_specification={}):
+    from parsl.executors.taskvine.utils import load_variable_in_serverless
+    y = load_variable_in_serverless('y')
+    return y + x + 1
+
+
+def f_local_compute(x):
+    return x + 1
+
+
+@require_taskvine
+@pytest.mark.taskvine
+@pytest.mark.parametrize('num_tasks', (1, 50))
+def test_function_context_computation(num_tasks, current_config_name):
+    if current_config_name != 'taskvine_ex':
```

Comment:
if you want to test against a specific configuration, have a look at tests that are `@pytest.mark.local` and have their own `with parsl.load()` in them - rather than using some ambient environment we don't expect the feature to work in.

That would be more consistent with existing tests. parsl/tests/test_monitoring/test_basic.py is a complicated example. or parsl/tests/test_htex/test_priority_queue.py

Reply:
I just pushed 3a079dd424 onto this branch to demonstrate what I think this should look like